reana_commons/config.py

```diff
@@ -336,18 +336,20 @@ def kubernetes_node_label_to_dict(node_label):
 SHARED_VOLUME_PATH = os.getenv("SHARED_VOLUME_PATH", "/var/reana")
 """Default shared volume path."""
 
-DEFAULT_WORKSPACE_PATH = os.getenv("DEFAULT_WORKSPACE_PATH", SHARED_VOLUME_PATH)
-"""Default workspace path defined by the admin."""
+WORKSPACE_PATHS = json.loads(os.getenv("WORKSPACE_PATHS", "{}"))
+"""Dictionary of available workspace paths with pairs of cluster_node_path:cluster_pod_mountpath."""
 
 
-def workspaces(workspaces_list):
-    """Load the available workspaces as list."""
-    workspaces_list = workspaces_list.split(",") if workspaces_list else []
-    return workspaces_list
+def default_workspace():
+    """Obtain default workspace path."""
+    workspaces_list = [
+        path for path in list(WORKSPACE_PATHS.values()) if path != SHARED_VOLUME_PATH
+    ]
+    return workspaces_list[0] if workspaces_list else SHARED_VOLUME_PATH
 
 
-WORKSPACE_PATHS = workspaces(os.getenv("WORKSPACE_PATHS"))
-"""List of allowed workspace paths."""
+DEFAULT_WORKSPACE_PATH = default_workspace()
+"""Default workspace path defined by the admin."""
```

Comment:
Please have a look into my comment: https://github.com/reanahub/reana/pull/538#discussion_r712870449

I think something like this would solve the problem:
```python
def workspaces(paths):
    if isinstance(paths, list):
        return dict(p.split(":") for p in paths)
    return paths


WORKSPACE_PATHS = workspaces(json.loads(os.getenv("WORKSPACE_PATHS", "{}")))
"""Dictionary of available workspace paths with pairs of cluster_node_path:cluster_pod_mountpath."""


def default_workspace():
    """Obtain default workspace path."""
    workspaces_list = list(WORKSPACE_PATHS.values())
    return workspaces_list[0] if workspaces_list else SHARED_VOLUME_PATH


DEFAULT_WORKSPACE_PATH = default_workspace()
"""Default workspace path defined by the admin."""
```

Let me know what do you think

Reply:
I though about it, but as I have to pass `WORKSPACE_PATHS` to the [jobs specification files](https://github.com/reanahub/reana-workflow-controller/pull/399/files) `{"name": "WORKSPACE_PATHS", "value": json.dumps(WORKSPACE_PATHS)}`, I find a little unnecessary to convert it to a list of strings again, pass it to the jobs and then transform it into a dictionary again. It wouldn't be better to directly pass the value?

Reply:
Yes, since you dump the json there, it will be only a string which could be passed as an envar and loaded here in `reana-commons` without any problems. That's why I've added `if isinstance(paths, list)` check to transform it to dict only when coming from Helm. For `workflow run manager` you can pass the dump of the json as it is now